server/service/mdm.go

```diff
@@ -1116,6 +1117,179 @@ func isAppleProfileID(profileIDOrUUID string) (uint, bool) {
 	return uint(id), true
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// POST /mdm/profiles (Create Apple or Windows MDM Config Profile)
+////////////////////////////////////////////////////////////////////////////////
+
+type newMDMConfigProfileRequest struct {
+	TeamID  uint
+	Profile *multipart.FileHeader
+}
+
+func (newMDMConfigProfileRequest) DecodeRequest(ctx context.Context, r *http.Request) (interface{}, error) {
+	decoded := newMDMConfigProfileRequest{}
+
+	err := r.ParseMultipartForm(512 * units.MiB)
+	if err != nil {
+		return nil, &fleet.BadRequestError{
+			Message:     "failed to parse multipart form",
+			InternalErr: err,
+		}
+	}
+
+	val, ok := r.MultipartForm.Value["team_id"]
+	if !ok || len(val) < 1 {
+		// default is no team
+		decoded.TeamID = 0
+	} else {
+		teamID, err := strconv.Atoi(val[0])
+		if err != nil {
+			return nil, &fleet.BadRequestError{Message: fmt.Sprintf("failed to decode team_id in multipart form: %s", err.Error())}
+		}
+		decoded.TeamID = uint(teamID)
+	}
+
+	fhs, ok := r.MultipartForm.File["profile"]
+	if !ok || len(fhs) < 1 {
+		return nil, &fleet.BadRequestError{Message: "no file headers for profile"}
+	}
+	decoded.Profile = fhs[0]
+
+	return &decoded, nil
+}
+
+type newMDMConfigProfileResponse struct {
+	ProfileID string `json:"profile_id"`
```

Comment:
heads-up @ghernandez345 that the `profile_id` of this endpoint is a string, and will be unmarshaled as such (for your typings and/or if you need to do some form of casting)